app/mailers/candidate_mailer.rb

```diff
@@ -541,7 +541,12 @@ def candidate_realistic_job_preview_link(candidate)
     realistic_job_preview_url({ id: candidate.pseudonymised_id }.merge(utm_args))
   end
 
-  helper_method :candidate_magic_link, :candidate_realistic_job_preview_link
+  def candidate_unsubscribe_link(candidate)
+    token = candidate.generate_token_for :unsubscribe_link
```

Comment:
Does this generates new tokens for each apply deadline email sent?

Maybe I understood wrong but I think there’s a potential issue with handling tokens when sending multiple emails. 

If a candidate receives multiple emails (one in July and one in August), each should have a unique token. 

If multiple emails are sent with the same unsubscribe token, using the token from one email will invalidate the link in all other emails. This could lead to a bug for the user.

Can you confirm, maybe is me that understood wrong.

Reply:
I explained badly above. 

An example what I meant:

Monday:
Email 1 is sent with link: https://example.com/unsubscribe?token=abc123.
Database token: abc123.

Wednesday:
Email 2 is sent with link: https://example.com/unsubscribe?token=def456.
Database token is updated to: def456.

Thursday:
Candidate clicks the link in Email 1: https://example.com/unsubscribe?token=abc123.
Since the database now only recognizes def456, abc123 is invalid, resulting in a 404 error.

Again maybe I understood wrong. Apologies if I did.

Reply:
The token doesn't expire and is not one-time use. It's not a magic link. You can test this locally if you like. 
- generate an unsubscribe link to for a user and follow it. The candidate will have `unsubscribed_from_emails: true`
- follow the link again. You will see the unsubscribe site successfully again. unsubscribed is still true
- change unsubscribed_from_emails to false.
- follow the link again. The link is still valid and the candidate is unsubscribed again.

Reply:
I think the tokens get generated on every hit and may be unique - they'll all resolve back to the same Candidate - Rails magic 🪄

Reply:
If is Rails magic (I need to double check the docs!), I'd feel more comfortable if we have one more scenario in the system/feature spec (like the example I sent above), just in case this feature changes in a future Rails version and we aren't caught by surprise on any unexpected bugs.

But everything else looks good!

Reply:
I can add some specs to demonstrate that this is safe so we don't have to worry about it.